src/pages/oauth-result.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { Center, Heading, Text } from "@chakra-ui/react"
+import { PlatformName } from "@guildxyz/types"
+import { GetServerSideProps, NextPage } from "next"
+import { useRouter } from "next/router"
+import platforms from "platforms/platforms"
+import { useEffect, useState } from "react"
+
+export type OAuthResultParams =
+  | {
+      status: "success"
+      platform: PlatformName
+      path: string
+    }
+  | {
+      status: "error"
+      message: string
+      platform?: PlatformName
+      path?: string
+    }
+
+const OAuth: NextPage<OAuthResultParams> = (query) => {
+  const router = useRouter()
+  const [hasReceivedConfirmation, setHasReceivedConfirmation] = useState(false)
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout
+    if (query.platform) {
+      const channel = new BroadcastChannel(`guild-${query.platform}`)
+      channel.onmessage = (event) => {
+        if (event.isTrusted && event.origin === window.origin) {
+          if (event.data?.type === "oauth-confirmation") {
```

Comment:
Since we don't do anything else in this block, could we move this condition inside the `if` above it? Just to reduce the number of nested blocks here, hehe

Reply:
Agreed, made this simplification in https://github.com/guildxyz/guild.xyz/pull/1113/commits/4e5d36e1f5ba0cb3b27205970684318ef8710cd5